### PR TITLE
feat: add ddl client

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add DDL client

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#1727

### CODE REVIEW

File: src/meta-client/src/client/ddl.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            .map(|url| url.as_ref().to_string())
+            .collect::<HashSet<_>>()
+            .drain()
+            .collect::<Vec<_>>();
+
+        Ok(())
+    }
+
+    // TODO(weny): considers refactoring `ask_leader` into a common client.
+    async fn ask_leader(&mut self) -> Result<()> {
```
Comment (MichaelScofield) [inline src/meta-client/src/client/ddl.rs:114]: can we make heartbeat client a field of this ddl client? so ddl client could just use the ask_leader method in heartbeat client

Comment (fengjiachun) [reply src/meta-client/src/client/ddl.rs:114]: Maybe we can add an API for HeartbeatClient: `heartbeat_client.get_leader() -> Peer` based on `ask_leader()`, then:
```
let leader = heartbeat_client.get_leader();
ddl.client.submit_task(leader);
```

**Comment by WenyXu** [comment]
@MichaelScofield @fengjiachun PTAL

**Review by MichaelScofield — APPROVED** [review]

**Comment by WenyXu** [comment]
PTAL @fengjiachun

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**